var/spack/repos/builtin/packages/scine-core/package.py

```diff
@@ -0,0 +1,42 @@
+# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+import os
+
+from spack.package import *
+
+
+class ScineCore(CMakePackage):
+    """Core provides the functionality necessary to couple the individual SCINE modules together"""
+
+    homepage = "https://scine.ethz.ch/"
+    url = "https://github.com/qcscine/core/archive/refs/tags/4.0.2.tar.gz"
+
+    version("4.0.2", "7181c6f93d71def22f1e0e5767afc7587c04b49abc03516f6926394868e7adc6")
+
+    resource(
+        name="dev",
+        url="https://github.com/qcscine/development-utils/archive/refs/tags/5.0.1.tar.gz",
+        sha256="089ca500fc191e04a968ea166d2fe80178b227bc2e6d3926523fa2eee5f6492d",
+        placement="_dev",
+    )
+
+    depends_on("boost+filesystem+program_options cxxstd=17 @1.65.0:")
+    depends_on("googletest")
+
+    def patch(self):
+        os.rmdir("dev")
+        os.rename("_dev", "dev")
+
+    def cmake_args(self):
+        return [
+            self.define("SCINE_BUILD_TESTS", self.run_tests),
+            self.define("SCINE_MARCH", ""),
```

Comment:
I'm not familiar with best practices for spack recipes but shouldn't this be set to the architecture that you are currently building on?

If the architecture is not forwarded here, I suspect it is possible that more "standard" ways of specifying the build architecture may not be respected (?).

Reply:
See https://spack.readthedocs.io/en/latest/basic_usage.html#support-for-specific-microarchitectures, spack has a way to inject the correct flags for the requested microarch. Having scine set them as well will interfere with this mechanism, therefore it is disabled here.

Reply:
Ok!

After looking into the cmake files, it appears indeed that this option is mostly used to prevent mixing of scine packages compiled for different architectures (which should not really be an issue with spack).
In some cases, it is use to set compile flags explicitly, which indeed is better handled by spack.

In one case it is used to forward this information to a dependency it is building (in the serenity wrapper https://github.com/qcscine/serenity_wrapper/blob/a2214474601ab4e75a9da35a1deb28ff6efe1874/CMakeLists.txt#L27, which is indeed not included here).
Also there, the correct solution is to have spack build the dependency rather than cmake.

All good!